src/telemetry.c

```diff
@@ -637,6 +640,71 @@ static int set_bios_version_header(struct telem_ref *t_ref)
         return status;
 }
 
+
+/**
+ * Generates an alphanumeric id of length len
+ *
+ * @param buff pointer to memory buffer
+ *
+ * @return 0 if successful, or -1 if there's a problem
+ *
+ */
+static int gen_event_id(char *buff)
+{
+        int frandom = -1;
+        int result = -1;
+        uint64_t random_id[2] = {0};
+
+        frandom = open("/dev/urandom", O_RDONLY);
+        if (frandom < 0) {
+                goto rand_err;
+        }
+
+        if (read(frandom, &random_id, sizeof(random_id)) != sizeof(random_id)) {
+                goto rand_err;
+        }
+
+        result = sprintf(buff, "%.16" PRIx64 "%.16" PRIx64, random_id[0], random_id[1]);
```

Comment:
I don't like getting a pointer here without some indication/protection of the buffer's length.